### DISCUSSION TITLE
Formatting rules

### DISCUSSION THREAD

**Discussion by BILLyTheLiTTle** [detekt/detekt#5035] [Q&A] [answered]
I am using Detekt in an Android project (from Android Studio)

### Attempt 1
I have read that Detekt uses Ktlint for code styling. I have generated `.editorConfig` but it seems that it is being ignored. 

### Attempt 2
I have  generated a Project.xml (IntellIJ Idea code style) but I cannot set it to Detekt as formatting ruleset.

### Attempt 3
I have read that exists a `detektIdeaFormat` but I cannot find it anywhere!

Can I make Detekt to respect my project's code styling?

**Comment by cortinico** [comment, marked as answer]
> ### Attempt 1
> I have read that Detekt uses Ktlint for code styling. I have generated `.editorConfig` but it seems that it is being ignored.

Nope we don't support `.editorconfig`. The reason is explained here #4975

> ### Attempt 2
> I have generated a Project.xml (IntellIJ Idea code style) but I cannot set it to Detekt as formatting ruleset.

We also don't support `Project.xml`. 

> ### Attempt 3
> I have read that exists a `detektIdeaFormat` but I cannot find it anywhere!
> Can I make Detekt to respect my project's code styling?

To my knowledge, the `detektIdeaFormat` task was removed long ago. Also could you share where you have read that? Is it in our official documentation?

To answer your original question: you could use our official IntelliJ Plugin:
https://plugins.jetbrains.com/plugin/10761-detekt

**Reply by BILLyTheLiTTle** [reply]
> To my knowledge, the detektIdeaFormat task was removed long ago. Also could you share where you have read that? Is it in our official documentation?

No, it exists in other articles and not in the official documentation.
Thanks
